urllib3/connection.py

```diff
@@ -162,6 +162,37 @@ def connect(self):
         conn = self._new_conn()
         self._prepare_conn(conn)
 
+    def request_chunked(self, method, url, body=None, headers={}):
+        """
+        Alternative to the common request method, which sends the
+        body with chunked encoding and not as one block
+        """
+        header_names = set(k.lower() for k in headers)
```

Comment:
"It's not used everywhere so there's no point using it here" is a terrible argument for not making a change. If you don't want to continue addressing feedback, that's fine. @Lukasa or I will finish this up for you.

I don't think the header dict adds more overhead in this case. On the contrary, it means we can deal with the headers in the method in an intuitive manner (relying on case-insensitivity when we need it and preserving case to send to the server when we need that.